Source/Manager/IdentityAPI/IdentityAPI.swift

```diff
@@ -229,6 +229,16 @@ class IdentityAPI {
         requestWithRetries(router: .updateProfile(userID: userID), formData: formData, headers: [.authorization: oauthToken.bearer], completion: completion)
     }
 
+    internal func updateUserDevice(
+        userID: String,
+        oauthToken: String,
+        device: UserDevice,
+        completion: @escaping ((Result<UserDevice, ClientError>) -> Void)
```

Comment:
As mentioned below `UserDevice` should be split in two, and this would then become something like `Result<UserDeviceHash, ClientError>`.